reach-whitepaper.md

```diff
@@ -0,0 +1,1019 @@
+interest) using compressed aggregated data structure admitting merge operation.
+Here we only consider one of the most basic families of sketches, called 
+[counting bloom filters (CBFs)](https://en.wikipedia.org/wiki/Counting_Bloom_filter). 
+In this case, we have a hash function $`h`$ that maps any user ID to one of $`m`$ 
+buckets. 
+The CBF sketch is simply the histogram on these $`m`$ buckets among all the 
+reached users. 
+We can estimate the reach based on the number of non-empty buckets in the 
+sketches. 
+The exact formula for this estimation is based on the hash function 
+distribution.
+
+The main advantage of this sketch is that it can be merged: if we have two CBFs,
+summing them up gives us the CBF corresponding to the union of the two
+(multi)sets.
+This allows us to be more flexible when answering queries.
+Recall that, in the direct method, we need to know all the queries beforehand
+since we need to allocate each bucket and each query. 
+This is not needed when using sketches: We can simply create a sketch for each
+fine-grained unit (e.g. the reach for each day) and, when we make a query for a
+day range, we take the union of all the sketches in that range.
+This means that the analyst can supply the queries in an on-demand fashion,
+which is more convenient. Moreover, sketching methodology offers another 
+significant advantage: the ability to deduplicate user IDs across multiple 
+devices, providing a more accurate picture of individual engagement. 
+This would typically require using the same user ID on multiple devices. 
+Shared Storage is partitioned per browser, so the ad tech would need to identify
+the user across devices independently. Alternatively, there are modeling-based
+approaches (e.g. 
+[virtual people](https://research.google/pubs/virtual-people-actionable-reach-modeling/))
+that would avoid the need for these
+IDs to map directly to real-world users.
+
+Nevertheless, there are also several disadvantages of this sketch.
+Firstly, each sketch is a histogram with $`m`$ buckets.
+In contrast, the direct method only outputs a single number per query.
+Therefore, the sketching method can result in a memory usage overhead during
+processing of the aggregatable reports and during post-processing of the summary
+reports compared to the direct or point contribution methods.
+Secondly, since the noise is added to every entry of the sketch, this can lead
+to inaccurate estimates if the noise is large.
+Specifically, as we will see below, we need to set the capping parameter to be
+small to ensure the noise is small but doing so results in a large error due to
+capping. 
+This tension means that we have to be very careful in terms of parameter 
+settings to ensure (reasonably) accurate estimates.
+
+The overall scheme of the reach estimation involving sketches is described on 
+the following diagram.
+
+![alt_text](figs/diagram.png "computation diagram")
```

Comment:
can we add a slightly more descriptive alt text?

Reply:
Added a more descriptive alt text.

Reply:
Ah sorry I meant more could we replace `alt_text` with something like `computation diagram` (although the longer title is good too!)

Reply:
Thanks for clarification. I used a more longer title and moved description in alt text.